cmd/antrea-agent/config.go

```diff
@@ -48,6 +48,9 @@ type AgentConfig struct {
 	// Make sure it doesn't conflict with your existing interfaces.
 	// Defaults to antrea-gw0.
 	HostGateway string `yaml:"hostGateway,omitempty"`
+	// Name of the interface antrea-agent will create and use for WireGuard traffic encryption.
+	// Defaults to antrea-wg0.
+	WireGuardInterfaceName string `yaml:"wireGuardInterfaceName,omitempty"`
```

Comment:
Sorry for the last minute suggestion. Want to know if anyone suggested to have this configuration? If not, can we remove it for now? Two reasons in my mind:
1. We just started grouping configs specific to a feature in a struct, I wonder if we should group wireGuardInterfaceName and wireGuardPort together. But if it's just single wireGuardPort, then it seems fine to have it in first level.
2. Although hostGateway is configurable, I didn't know anyone changed it. And we didn't make other interface names configurable after that, for example, antrea-egress0. Having it configurable will add some extra burdun to our code, for instance, when we want to exclude antrea managed devices when searching for host managed devices. So I would prefer not exposing it for now and we can consider when there is a real requirement for it.

cc @jianjuns @antoninbas for opinions.